lib/reform/form/active_model/result.rb

```diff
@@ -0,0 +1,15 @@
+class Reform::Contract::Result  
+  private
+
+  def filter_for(method, *args)
+    @results.collect { |r| r.public_send(method, *args).to_h }
+            .inject({}) { |hah, err| hah.merge(err) { |key, old_v, new_v| (new_v.is_a?(Array) ? (old_v |= new_v) : old_v.merge(new_v)) } }
+            .find_all { |k, v| # filter :nested=>{:something=>["too nested!"]} #DISCUSS: do we want that here?
+              if v.is_a?(Hash)
+                nested_errors = v.select { |attr_key, val| attr_key.is_a?(Integer) && val.is_a?(Array) && val.any? }
+                v = nested_errors.to_a if nested_errors.any?
+              end
+              v.is_a?(Array) || v.class.to_s == "ActiveModel::DeprecationHandlingMessageArray"
```

Comment:
I did similar as a monkey patch to get our 6.1 upgrade in. Curious why v.class.to_s vs `v.is_a?(ActiveModel::DeprecationHandlingMessageArray)`?

is_a?() seems more direct than stringifying the class name and comparing the string.

Reply:
`v.is_a?(ActiveModel::DeprecationHandlingMessageArray` doesn' work on Rails <= 6.0. It will raise `NameError`.